apollo-router/src/spec/query.rs

```diff
@@ -927,6 +906,36 @@ impl Query {
         Ok(())
     }
 
+    fn has_only_typename_field(&self, selection_set: &Vec<Selection>, variables: &Object) -> bool {
```

Comment:
I need this function because of: https://github.com/apollographql/router/pull/6009/files?diff=split&w=1#r1762118849

Note in order to address point 2 from above comment I need to skip differed fields